src/ttf.cpp

```diff
@@ -149,6 +155,39 @@ bool TtfFont::LoadFromFile(FT_Library fontLibrary, bool nameOnly) {
         return false;
     }
 
+    return PostLoadProcess(nameOnly);
+}
+
+//-----------------------------------------------------------------------------
+// Load a TrueType from resource in memory. Implemented to load bundled fonts
+// through theresource system.
+//-----------------------------------------------------------------------------
+bool TtfFont::LoadFromResource(FT_Library fontLibrary) {
+    size_t _size;
+    const void *_buffer = Platform::LoadResource(fontFile.raw, &_size);
+
+    FT_Long size = static_cast<FT_Long>(_size);
+    const FT_Byte *buffer = reinterpret_cast<const FT_Byte*>(_buffer);
+
+    if(int fterr = FT_New_Memory_Face(fontLibrary, buffer, size, 0, &fontFace)) {
+            dbp("freetype: loading font '%s' from memory failed: %s",
+                fontFile.raw.c_str(), ft_error_string(fterr));
+            return false;
+    }
+
+    // Call with nameOnly = false as a quick hack to prevent the font from being
+    // lazily loaded as a file. This could also be implemented via a `bool
+    // is_resource` member in the TtfFont class. However, as long as there are only
+    // few bundled fonts, this works fine.
```

Comment:
I agree that this is suboptimal. Do you think you could add a prefix for builtin fonts, e.g. `res:`, so that the font you added becomes `res:/fonts/BitstreamVeraSans-Roman-builtin.ttf`? This would also unambiguously distinguish builtin fonts in savefiles (so that system fonts with the same filename don't interfere with them), and also allow us to add new builtin fonts and keep compatibility between versions.

Reply:
Sure, that makes sense, although I would change it to 'res://' similar to protocol prefixes such as 'ssh://' etc. I feel like that makes the intent very clear.